File: packages/react-native-autoplay/android/src/main/java/com/margelo/nitro/at/g4rb4g3/autoplay/hybrid/HybridListTemplate.kt
package com.margelo.nitro.at.g4rb4g3.autoplay.hybrid

import com.margelo.nitro.at.g4rb4g3.autoplay.AndroidAutoSession
import com.margelo.nitro.at.g4rb4g3.autoplay.template.AndroidAutoTemplate
import com.margelo.nitro.at.g4rb4g3.autoplay.template.ListTemplate

class HybridListTemplate : HybridHybridListTemplateSpec() {

    override fun createListTemplate(config: ListTemplateConfig) {
        val context = AndroidAutoSession.Companion.getRootContext()
            ?: throw IllegalArgumentException("createListTemplate failed, carContext not found")

        val template = ListTemplate(context, config)
        AndroidAutoTemplate.Companion.setTemplate(config.id, template)
    }

    override fun updateListTemplateSections(
        templateId: String, sections: Array<NitroSection>?
    ) {
        val template = AndroidAutoTemplate.Companion.getTemplate<ListTemplate>(templateId)
        template.updateSections(sections)
    }
}

File: packages/react-native-autoplay/ios/hybrid/HybridListTemplate.swift
//
//  HybridListTemplate.swift
//  Pods
//
//  Created by Manuel Auer on 15.10.25.
//

import NitroModules

class HybridListTemplate: HybridListTemplateSpec {
    func createListTemplate(config: ListTemplateConfig) throws {
        let template = ListTemplate(config: config)
        TemplateStore.addTemplate(
            template: template,
            templateId: config.id
        )
    }

    func updateListTemplateSections(
        templateId: String,
        sections: [NitroSection]?
    ) throws -> Promise<Void> {
        return Promise.async {
            guard
                let template = TemplateStore.getTemplate(templateId: templateId)
                    as? ListTemplate
            else {
                throw AutoPlayError.invalidTemplateType(
                    "\(templateId) is not a ListTemplate"
                )
            }

            await template.updateSections(sections: sections)
        }
    }
}
